### PR TITLE
Cleaned up dependencies in pom files

### PR DESCRIPTION
**Pull request by cdeszaq**
Note: Based on top of the final v0.5.x commit. Once that merges, this PR will be re-pointed to master.

This PR cleans up pom dependencies so that the parent pom manages the specific versions of all of the dependencies the child poms have. It also cleans up the parent pom's `dependencies` section so that child poms are only inheriting dependencies they _all_ should have (ie. Spock, logging, etc.)

This also bumps the version to v0.6.x.

### CODE REVIEW

File: fili-core/pom.xml [hunk 1]
```diff
@@ -29,230 +28,185 @@
         <dependency>
             <groupId>com.yahoo.fili</groupId>
             <artifactId>fili-system-config</artifactId>
-            <version>0.1-SNAPSHOT</version>
```
Comment (archolewa) [inline fili-core/pom.xml]: The version information is now being picked up from the root `pom.xml` yes?

File: pom.xml [hunk 2]
```diff
@@ -114,91 +115,378 @@
...
+            </dependency>
+
+            <!-- All things Spring -->
+            <dependency>
+                <groupId>org.springframework</groupId>
+                <artifactId>spring-core</artifactId>
+                <version>4.0.5.RELEASE</version>
+            </dependency>
+
+            <!-- Test -->
```
Comment (archolewa) [inline pom.xml]: We should probably move all of the Spock (and maybe Groovy?) dependencies under this "section" as well, since those are for testing.

**Comment by archolewa** [comment]
👍

File: pom.xml (hunk 2 above)
Comment (cdeszaq) [inline pom.xml]: Spock definitely is (I'll make that change), but Groovy might not be... 😉

Comment (cdeszaq) [inline pom.xml]: Updated

**Comment by RVRSKumar** [comment]
👍

**Comment by cdeszaq** [comment]
Merged

**Pull request closed without merging**